src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NativeLayoutVertexNode.cs

```diff
@@ -83,60 +83,135 @@ protected Vertex SetSavedVertex(Vertex value)
         }
     }
 
-    internal sealed class NativeLayoutMethodLdTokenVertexNode : NativeLayoutSavedVertexNode
+    internal abstract class NativeLayoutMethodEntryVertexNode : NativeLayoutSavedVertexNode
     {
-        private MethodDesc _method;
+        protected readonly MethodDesc _method;
+        private bool _createInstantiatedSignature;
+        private bool _saveEntryPoint;
         private NativeLayoutTypeSignatureVertexNode _containingTypeSig;
         private NativeLayoutMethodSignatureVertexNode _methodSig;
         private NativeLayoutTypeSignatureVertexNode[] _instantiationArgsSig;
 
-        protected override string GetName() => "NativeLayoutMethodLdTokenVertexNode_" + NodeFactory.NameMangler.GetMangledMethodName(_method);
-
-        public NativeLayoutMethodLdTokenVertexNode(NodeFactory factory, MethodDesc method)
+        public NativeLayoutMethodEntryVertexNode(NodeFactory factory, MethodDesc method, bool createInstantiatedSignature, bool saveEntryPoint)
```

Comment:
1. ok
2. Implementing on the derived type makes it more complicated IMHO. I want this node to return the single vertex constructed by the writer.GetMethodSignature API, which captures the entry point info. The reason we have the entry point as an optional component is that this vertex type is also used by the RuntimeMethodHandle signatures, which doesn't really care about the entry point (and in PN we omitted it to save size on disk I guess, and not grow the external references table unnecessarily...)

---
In reply to: [97848095](https://github.com/dotnet/corert/pull/2575#discussion_r97848095) [](ancestors = 97848095)